lms/lib/xblock/test/test_mixin.py

```diff
@@ -130,6 +132,49 @@ def test_validate_invalid_groups(self):
             ValidationMessage.ERROR,
         )
 
+    def test_validate_nonsensical_access_restriction(self):
```

Comment:
Please add a couple of new test cases to cover the logic changes you made (all groups must be a subset of the parent unit groups, a component that is set to "All Learners and Staff" displays an error if the unit has restricted access.